src/Functions/reinterpretAs.cpp

```diff
@@ -216,7 +222,13 @@ class FunctionReinterpret : public IFunction
 
                     for (size_t i = 0; i < size; ++i)
                     {
-                        memcpy(&vec_res[i], &data_from[offset], copy_size);
+                        if constexpr (std::endian::native == std::endian::little)
+                            memcpy(&vec_res[i], &data_from[offset], copy_size);
+                        else
+                        {
+                            size_t offset_to = sizeof(ToFieldType) > copy_size ? sizeof(ToFieldType) - copy_size : 0;
+                            memcpy(reinterpret_cast<char*>(&vec_res[i]) + offset_to, &data_from[index - offset], copy_size);
```

Comment:
Apparently this should also use `reverseMemcpy` as the case for `String` - https://github.com/ClickHouse/ClickHouse/pull/76253